### PR TITLE
rust cli: Bring ros1 bag convert to parity

### PR DESCRIPTION
**Pull request by amacneil**
### Changelog

None

### Docs

None

### Description

Rust `mcap convert` now emits ROS1 schemas and channels when ROS bag connection records are read, then writes messages to those known channel IDs. This matches Go CLI behavior more closely and preserves ROS `conn` IDs even for channels that have no messages.

Test coverage now includes:

- Synthetic ROS1 bags for empty connections, duplicate channel content with distinct IDs, duplicate connection records inside chunks, and conflicting duplicate connection records.
- Noetic-generated ROS1 bags in `testdata/bags/generated/` covering uncompressed, bz2, lz4, and empty bags.
- Rust command-level conversion assertions for topics, schema names, timestamps, channel mapping, and payload bytes.
- CLI conformance cases comparing Go and Rust conversion output for the Noetic uncompressed, bz2, and lz4 bags.

<div><a href="https://cursor.com/agents/bc-11f78ac3-011a-452e-b209-40b3c4c15993"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-11f78ac3-011a-452e-b209-40b3c4c15993"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by amacneil** [comment]
> One open question on regeneration: does rosbag.Bag actually produce byte-identical output across runs on the same host? 

It appears to be deterministic, I ran it several times. Not sure whether it will be identical across machines though.

File: testdata/bags/generate_noetic_bags.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  --workdir /workspace/testdata/bags \
+  "$image" \
+  bash -lc "
+    set -euo pipefail
+    apt-get update
+    apt-get install -y --no-install-recommends ros-noetic-roslz4 ros-noetic-std-msgs
+    rm -rf /var/lib/apt/lists/*
+    source /opt/ros/noetic/setup.bash
+    python3 generate_noetic_bags.py --output '$output_dir'
+    chown -R '$host_uid:$host_gid' '$output_dir'
```
[the remark this answers is not in the document]
Comment (amacneil) [reply testdata/bags/generate_noetic_bags.sh:22]: this is overly pedantic, it's fine

File: testdata/bags/generate_noetic_bags.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        "--output",
+        type=Path,
+        default=Path("generated"),
+        help="directory to recreate and write generated .bag fixtures into",
+    )
+    args = parser.parse_args()
+
+    if args.output.exists():
+        shutil.rmtree(args.output)
+    args.output.mkdir(parents=True)
```
[the remark this answers is not in the document]
Comment (amacneil) [reply testdata/bags/generate_noetic_bags.py:53]: yeah, that's intentional. we want to delete the generated directory before regenerating the files inside it.

File: rust/cli/src/commands/convert/ros1_bag.rs [hunk 3]
```diff
@@ -730,6 +918,131 @@ mod tests {
...
+            input,
+            mcap::WriteOptions::new().profile("ros1").compression(None),
+        )
+        .expect_err("conflicting duplicate connection should fail");
+
+        assert!(err.to_string().contains("failed to write ROS1 channel"));
+        assert!(err
+            .root_cause()
+            .to_string()
+            .contains("multiple records that don't match"));
```
[the remark this answers is not in the document]
Comment (amacneil) [reply rust/cli/src/commands/convert/ros1_bag.rs:1043]: It’s all in the same repo, if it breaks we can fix it

File: testdata/bags/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+testdata/bags/generate_noetic_bags.sh
+```
+
+The script runs the official `ros:noetic-ros-base` Docker image, installs the
+small message/compression packages needed by the generator, recreates
+`testdata/bags/generated/`, and writes regenerated bags there.
+
+Parser edge cases that cannot be produced through the public `rosbag.Bag` writer
+API, such as a connection record with no messages, should stay as small
+programmatic fixtures in the relevant tests.
```
[the remark this answers is not in the document]
Comment (amacneil) [reply testdata/bags/README.md:24]: This will figure it out soon enough
